angular-ngrx-scss/src/app/shared/components/repo-controls/repo-controls.component.ts

```diff
@@ -1,11 +1,43 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { FormControl } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { debounceTime, distinctUntilChanged, combineLatest } from 'rxjs';
+import { startWith } from 'rxjs/operators';
+import { setSortAndFilterProperties } from 'src/app/state/profile/profile.actions';
+import {
+  hasActiveSortAndFilters,
+  selectFilterBySearch,
+  selectReposCount,
+} from 'src/app/state/profile/profile.selectors';
+import { SortAndFilterState } from 'src/app/state/profile/profile.state';
 
 @Component({
   selector: 'app-repo-controls',
   templateUrl: './repo-controls.component.html',
   styleUrls: ['./repo-controls.component.scss'],
 })
-export class RepoControlsComponent {
+export class RepoControlsComponent implements OnInit {
   searchInput = new FormControl('');
+  hasActiveSortAndFilters$ = this.store.select(hasActiveSortAndFilters);
+  selectReposCount$ = this.store.select(selectReposCount);
+
+  selectFilterBySearch$ = this.store.select(selectFilterBySearch);
+  constructor(private store: Store) {}
+
+  ngOnInit(): void {
+    const searchInput$ = this.searchInput.valueChanges.pipe(
+      debounceTime(400),
+      distinctUntilChanged(),
+      startWith(''),
+    );
+
+    combineLatest([searchInput$], (text) => ({
+      search: text,
+      type: '',
+      language: '',
+      sort: '',
+    })).subscribe((criteria: SortAndFilterState) => {
+      this.store.dispatch(setSortAndFilterProperties({ filters: criteria }));
+    });
```

Comment:
blocker(memory-leak): The subscription is not cleaned up when the component is destroyed. We should either call `unsubscribe()` in `ngOnDestroy` or use a `destroy$` subject. The later approach is my preferred method personally, but either would work.